homeassistant/components/notify/html5.py

```diff
@@ -320,12 +330,29 @@ def check_authorization_header(self, request):
 class HTML5NotificationService(BaseNotificationService):
     """Implement the notification service for HTML5."""
 
-    def __init__(self, gcm_key, registrations, json_path):
+    def __init__(self, hass, gcm_key, registrations, json_path):
         """Initialize the service."""
         self._gcm_key = gcm_key
         self.registrations = registrations
         self.registrations_json_path = json_path
 
+        async def async_dismiss_message(service):
+            """Handle dismissing notification message service calls."""
+            kwargs = {}
+
+            if self.targets is not None:
+                kwargs[ATTR_TARGET] = self.targets
+            elif service.data.get(ATTR_TARGET) is not None:
+                kwargs[ATTR_TARGET] = service.data.get(ATTR_TARGET)
+
+            kwargs[ATTR_DATA] = service.data.get(ATTR_DATA)
+
+            await self.async_dismiss(**kwargs)
+
+        hass.services.async_register(
```

Comment:
This is not within an async context so we shouldn't be using the home assistant async api. Use `hass.services.register`.

Also, we try to avoid side effects in init method.